src/exec.cc

```diff
@@ -4,18 +4,33 @@
 #include <stdexcept>
 #include <string>
 #include <array>
+#include <tuple>
+#include <sys/wait.h>
+#include <cstring>
 
-std::string execute(const std::string & command) {
+std::tuple<std::string, int> execute(const std::string & command) {
     const char* cmd = command.c_str();
     std::array<char, 128> buffer;
-    std::string result;
-    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
+    std::string output;
+    FILE* pipe = popen(cmd, "r");
     if (!pipe) {
         throw std::runtime_error("popen() failed!");
     }
-    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
-        result += buffer.data();
+    while (fgets(buffer.data(), buffer.size(), pipe) != nullptr) {
+        output += buffer.data();
     }
-    return result;
+    
+    int exit_code = pclose(pipe);
+    if (WIFSIGNALED(exit_code) || WIFSTOPPED(exit_code)) {
+        throw std::runtime_error(std::string("process received signal ")
```

Comment:
Fun fact: if `/bin/sh` is Bash on your system, this never actually gets hit, because `popen` executes the command through `/bin/sh`, and Bash doesn't segfault itself in response to the child process segfaulting _only when you're redirecting stderr_, it just `exit()`s normally returning `-signal`. I have no idea what's going on there, or whether that behavior is supposed to be standardized.